subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/projectmodule/DefaultProjectDependencyPublicationResolver.java

```diff
@@ -41,28 +43,28 @@
 public class DefaultProjectDependencyPublicationResolver implements ProjectDependencyPublicationResolver {
     private final ProjectPublicationRegistry publicationRegistry;
     private final ProjectConfigurer projectConfigurer;
+    private final ProjectStateRegistry projects;
 
-    public DefaultProjectDependencyPublicationResolver(ProjectPublicationRegistry publicationRegistry, ProjectConfigurer projectConfigurer) {
+    public DefaultProjectDependencyPublicationResolver(ProjectPublicationRegistry publicationRegistry, ProjectConfigurer projectConfigurer, ProjectStateRegistry projects) {
         this.publicationRegistry = publicationRegistry;
         this.projectConfigurer = projectConfigurer;
+        this.projects = projects;
     }
 
     @Override
-    public <T> T resolve(Class<T> coordsType, ProjectDependency dependency) {
+    public <T> T resolve(Class<T> coordsType, Path identityPath) {
+        ProjectState projectState = projects.stateFor(identityPath);
         // Could probably apply some caching and some immutable types
 
-        ProjectInternal dependencyProject = (ProjectInternal) dependency.getDependencyProject();
-        return resolve(coordsType, dependencyProject);
-    }
-
-    @Override
-    public <T> T resolve(Class<T> coordsType, ProjectInternal project) {
-
         // Ensure target project is configured
-        projectConfigurer.configureFully(project);
+        projectConfigurer.configureFully(projectState);
```

Comment:
💭 I think you can inject a `ProjectRegistry` to this type and you can get a `ProjectInternal` from it for the path to pass to the configurer, if you change it back to configuring `ProjectInternal`s.

Reply:
In general, with the coming project isolation changes, we should avoid interacting with individual `Project/ProjectInternal` instances and instead interact with their owning `ProjectState` instances. 

Though I've reverted most of the other `ProjectConfigurer` code, this code here is the only place we call `configureFully`, so I feel it is safe to refactor this code to only require the owning `ProjectState`. The implementation of `configureFully` always called `getOwner`, and does not require the owned `ProjectInternal` instance.